packages/workbox-core/test/bundle/node/utils/test-logHelper.js

```diff
@@ -0,0 +1,339 @@
+import chai from 'chai';
+import sinon from 'sinon';
+
+import logHelper from '../../../../src/utils/logHelper';
+
+const expect = chai.expect;
+
+describe(`logHelper [${process.env.NODE_ENV}]`, function() {
+  let stubs = [];
+
+  beforeEach(function() {
+    logHelper.logLevel = logHelper.LOG_LEVELS.verbose;
+  });
+
+  afterEach(function() {
+    stubs.forEach((stub) => {
+      stub.restore();
+    });
+
+    stubs = [];
+  });
+
+  describe(`LOG_LEVELS`, function() {
+    it(`should expose the valid LOG_LEVELS`, function() {
+      expect(logHelper.LOG_LEVELS).to.exist;
+    });
+
+    it(`should expose the expected LOG_LEVELS`, function() {
+      expect(logHelper.LOG_LEVELS.verbose).to.exist;
+      expect(logHelper.LOG_LEVELS.debug).to.exist;
+      expect(logHelper.LOG_LEVELS.warning).to.exist;
+      expect(logHelper.LOG_LEVELS.error).to.exist;
+    });
+  });
+
+  describe(`.logLevel (setter)`, function() {
+    it(`should allow valid log levels`, function() {
+      expect(() => {
+        logHelper.logLevel = logHelper.LOG_LEVELS.verbose;
+        logHelper.logLevel = logHelper.LOG_LEVELS.debug;
+        logHelper.logLevel = logHelper.LOG_LEVELS.warning;
+        logHelper.logLevel = logHelper.LOG_LEVELS.error;
+      }).to.not.throw();
+    });
+
+    // TODO: Catch WorkboxError with error code
+    it(`should not allow log level less than verbose`, function() {
+      expect(() => {
+        logHelper.logLevel = logHelper.LOG_LEVELS.verbose - 1;
+      }).to.throw();
+    });
+
+    // TODO: Catch WorkboxError with error code
+    it(`should not allow log level greater than error`, function() {
+      expect(() => {
+        logHelper.logLevel = logHelper.LOG_LEVELS.error + 1;
+      }).to.throw();
+    });
+
+    // TODO: Catch WorkboxError with error code
+    it(`should not allow non-number log levels`, function() {
+      expect(() => {
+        logHelper.logLevel = undefined;
+      }).to.throw();
+
+      expect(() => {
+        logHelper.logLevel = null;
+      }).to.throw();
+
+      expect(() => {
+        logHelper.logLevel = '';
+      }).to.throw();
+
+      expect(() => {
+        logHelper.logLevel = [];
+      }).to.throw();
+
+      expect(() => {
+        logHelper.logLevel = {};
+      }).to.throw();
+    });
+  });
+
+  /**
+   * Why .calledWithMatch()?
+   *
+   * This method on sinon spies will ensure that the spied method was called
+   * "with matching arguments (and possibly others)." This means logHelper
+   * can add a prefix to the log and still pass the assertion.
+   */
+
+  describe('.log()', function() {
+    it('should work without input', function() {
+      const stub = sinon.spy(console, 'log');
+      stubs.push(stub);
+
+      logHelper.log();
+
+      expect(console.log.callCount).to.equal(1);
```

Comment:
Since you're creating a spy object, I'd kind of expect the check to be done against `spy` rather than against the method that's being spied on.

Apparently they're equivalent, and I don't very strongly, but the following reads better to me:

```js
expect(spy.calledOnce).to.be.true;
expect(spy.calledWithMatch('', 'test', null, undefined, [], {})).to.be.true;
```